packages/core/src/internal/base/button.base.ts

```diff
@@ -129,14 +129,19 @@ export class CdsBaseButton extends LitElement {
     }
 
     this.readonly = this.readonly || this.isAnchor;
-    this.role = this.readonly ? null : 'button';
 
     if (this.readonly) {
+      this.role = null;
       this.tabIndexAttr = null;
       this.ariaDisabled = null;
     } else {
+      this.role = 'button';
       this.tabIndexAttr = this.disabled ? -1 : 0;
       this.ariaDisabled = this.disabled ? 'true' : 'false';
     }
+
+    if (props.has('readonly')) {
+      this.setupNativeButtonBehavior(this.readonly);
+    }
```

Comment:
The updating of attributes that happens in the first if/else goes through two `updated` cycles in Lit Element to complete. I'm not sure if some of the attributes have unique properly that makes it so, but this made it impossible to combine this if/else with the separate logic right below it (setting up the native button behavior) since the condition for `props.has('readonly')` was false in the second cycle resulting in inconsistent / incorrect behavior.

Reply:
that's possibly caused by the update of the properties in this function.

Reply:
i ran into something similar with animations where i updated a LitElement property and it almost immediately went through another update loop.